playkit/src/main/java/com/kaltura/playkit/PKMediaEntry.java

```diff
@@ -86,13 +88,21 @@ public int describeContents() {
     public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(this.id);
         dest.writeString(this.name);
-        dest.writeTypedList(this.sources);
+        if (sources != null) {
+            dest.writeTypedList(this.sources);
+        } else {
+            dest.writeTypedList(Collections.EMPTY_LIST);
+        }
         dest.writeLong(this.duration);
         dest.writeInt(this.mediaType == null ? -1 : this.mediaType.ordinal());
-        dest.writeInt(this.metadata.size());
-        for (Map.Entry<String, String> entry : this.metadata.entrySet()) {
-            dest.writeString(entry.getKey());
-            dest.writeString(entry.getValue());
+        if (this.metadata != null) {
+            dest.writeInt(this.metadata.size());
+            for (Map.Entry<String, String> entry : this.metadata.entrySet()) {
+                dest.writeString(entry.getKey());
+                dest.writeString(entry.getValue());
+            }
+        } else {
+            dest.writeBundle(new Bundle());
```

Comment:
in the PKMediaEntry(Parcel in) constructor after reading the int of metadata map size if the value is -1 don't
create the map at all leave the field null.